astropy/io/fits/file.py

```diff
@@ -561,11 +578,11 @@ def _open_filename(self, filename, mode, overwrite):
         # Make certain we're back at the beginning of the file
         # BZ2File does not support seek when the file is open for writing, but
         # when opening a file for write, bz2.BZ2File always truncates anyway.
-        if not (isinstance(self._file, bz2.BZ2File) and mode == 'ostream'):
+        if not (_is_bz2file(self._file) and mode == 'ostream'):
             self._file.seek(0)
 
     @classproperty(lazy=True)
-    def _mmap_available(cls):
+    def _mmap_available(self):
```

Comment:
Why this change ?

Reply:
That function had been flagged by flake8 as
`N805 first argument of a method should be named 'self'`
but like the rest of the second commit, it is not strictly related.

Reply:
Ok, but this `classproperty` thing is like a classmethod, so it should take a class as argument.

Reply:
So the `@classproperty` decorator takes precedence? Could add a `# noqa: N805` comment to silence flake then.

Reply:
Yes, it's a class method, not an instance one. 
What are this `N*` error numbers, I don't get this with flake8 ?

Reply:
```
In [3]: fits.file._File._mmap_available                                                                                                                                                        
Out[3]: True
```

Reply:
That's what I got from the command line:
```
> flake8 astropy/io/fits/file.py                                                                                          
astropy/io/fits/file.py:134:80: E501 line too long (80 > 79 characters)
astropy/io/fits/file.py:139:80: E501 line too long (80 > 79 characters)
astropy/io/fits/file.py:140:80: E501 line too long (80 > 79 characters)
astropy/io/fits/file.py:161:80: E501 line too long (82 > 79 characters)
astropy/io/fits/file.py:330:80: E501 line too long (80 > 79 characters)
astropy/io/fits/file.py:336:80: E501 line too long (81 > 79 characters)
astropy/io/fits/file.py:343:80: E501 line too long (84 > 79 characters)
astropy/io/fits/file.py:444:80: E501 line too long (81 > 79 characters)
astropy/io/fits/file.py:585:26: N805 first argument of a method should be named 'self'
```

Reply:
It seems to come from an additional plugin: https://pypi.org/project/pep8-naming/

Reply:
I agree with @saimn that this change should be undone.

Since we don't fail o N805 there's no reason for a noqa, right?

Reply:
Yes, the notifications are from the pep8 plugin, but that's part of the standard flake8 installation, at least with the Fink version and Conda. But if it is just a bug in flake8 I agree it should be reverted.

Reply:
So we revert it to `self` without noqa?